src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -738,7 +771,19 @@ void BaseCouplingScheme::doImplicitStep()
   } else {
     // no convergence achieved for the coupling iteration within the current time window
     if (_acceleration) {
+      // Load from storage into buffer
+      for (auto &pair : getAccelerationData()) {
+        const auto stamples = pair.second->stamples();
+        PRECICE_ASSERT(stamples.size() > 0);
+        pair.second->sample() = stamples.back().sample;
+      }
+
       _acceleration->performAcceleration(getAccelerationData());
+
+      // Store from buffer
+      for (auto &pair : getAccelerationData()) {
+        pair.second->setSampleAtTime(time::Storage::WINDOW_END, pair.second->sample());
+      }
```

Comment:
Future work: we will need all samples in acceleration.

Reply:
Added a note here. 9d354ebc707164039834d42399e400b5431de2ff